pkg/upgrade/control_plane.go

```diff
@@ -1042,7 +1059,11 @@ func (u *ControlPlaneUpgrader) updateInfrastructureReference(replacementKey ctrl
 	// prep the replacement
 	infra.SetResourceVersion("")
 	infra.SetName(replacementKey.Name)
-	infra.SetOwnerReferences(nil)
+
+	// Remove only the Machine owner reference so we preserve other refs
+	ownerRefs := getAllButMachineOwners(infra.GetOwnerReferences())
+	infra.SetOwnerReferences(ownerRefs)
```

Comment:
Is there a test for this change?

Reply:
Same here: taking it back for the lack of existing unit tests for `updateInfrastructureReference()`